adal/src/main/java/com/microsoft/aad/adal/AuthenticationException.java

```diff
@@ -86,6 +139,73 @@ public String getMessage() {
         return getLocalizedMessage(null);
     }
 
+    /**
+     * Gets the status code returned from http layer.
+     *
+     * @return status code from http layer. Return -1 if status code is not initialized.
+     */
+    public int getServiceStatusCode() {
+        return mServiceStatusCode;
+    }
+
+    /**
+     * Gets the response body that may be returned by the service.
+     *
+     * @return response body map, null if not initialized.
+     */
+    public HashMap<String, String> getHttpResponseBody() {
```

Comment:
Let's change this to `public Map<String, String> getHttpResponseBody()` - more interface  vs concrete stuff :)

Reply:
Same above. We need to parse it into json object when passing it between apps with intent bundle.